CompanyEmployees.Application/Handlers/CheckInAttendanceHandler.cs

```diff
@@ -0,0 +1,43 @@
+using AutoMapper;
+using CompanyEmployees.Application.Commands;
+using CompanyEmployees.Core.Domain.Entities;
+using CompanyEmployees.Core.Domain.Exceptions;
+using CompanyEmployees.Core.Domain.Repositories;
+using MediatR;
+using Shared.DataTransferObjects;
+
+namespace CompanyEmployees.Application.Handlers;
+
+internal sealed class CheckInAttendanceHandler : IRequestHandler<CheckInAttendanceCommand, AttendanceDto>
+{
+    private readonly IRepositoryManager _repository;
+    private readonly IMapper _mapper;
+
+    public CheckInAttendanceHandler(IRepositoryManager repository, IMapper mapper)
+    {
+        _repository = repository;
+        _mapper = mapper;
+    }
+
+    public async Task<AttendanceDto> Handle(CheckInAttendanceCommand request, CancellationToken cancellationToken)
+    {
+        // Check if employee already checked in today
```

Comment:
Consider adding a guard clause for `request.Attendance` being null to fail fast and avoid potential NullReferenceExceptions.

**Suggested change**
```suggestion
if (request.Attendance is null)
    throw new ArgumentNullException(nameof(request.Attendance));
```